### PR TITLE
Optimize Assignment Fetching

### PR DESCRIPTION
**Pull request by kobybum**
### Skip the first commit since it's only formatting

Fetch all moodle resources once instead of once per section.
The function now takes a few seconds instead of around 2 minutes for courses with many sections.

### CODE REVIEW

**Review by maxandron — COMMENTED** [review]
Cool library! (black)
Go crazy! run it on everything.
Just make sure to put the commits in `.git-blame-ignore-revs` 
(https://github.com/psf/black#migrating-your-code-style-without-ruining-git-blame)

This looks cool too
https://github.com/psf/black#github-actions

And this:
https://github.com/psf/black#show-your-style

:)

File: moodler/sections.py [hunk 1]
```diff
@@ -152,55 +153,49 @@ def get_assignments_by_section(course_id, sections_names=None, assignments_names
...
-        # be looking at, unless the list is None
-        if sections_names is not None:
-            if section["name"] not in sections_names:
-                continue
+        missing_assignments = set(assignments_names).difference(found_assignments)
+        if missing_assignments:
+            logger.error("Could not find the following assignments: %s", missing_assignments)
 
-            sections_not_found.remove(section["name"])
+    assignment_id_to_section = defaultdict(list)
```
Comment (maxandron) [inline moodler/sections.py]: Shouldn't this return `str` by default?
I got a little lost here
I understand why `assignments_by_section` is list by default, but why `assignment_id_to_section` is list I didn't understand

Comment (maxandron) [inline moodler/sections.py]: Shouldn't this return str by default?
I got a little lost here
I understand why assignments_by_section is list by default, but why assignment_id_to_section is list I didn't understand

Comment (kobybum) [reply moodler/sections.py]: You're right, it should just be a dict

**Pull request merged**